neo4j/_async/bookmark_manager.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) "Neo4j"
+# Neo4j Sweden AB [https://neo4j.com]
+#
+# This file is part of Neo4j.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import annotations
+
+import typing as t
+from collections import defaultdict
+
+from .._async_compat.concurrency import AsyncCooperativeLock
+from .._async_compat.util import AsyncUtil
+from ..api import (
+    AsyncBookmarkManager,
+    Bookmarks,
+)
+
+
+T_BmSupplier = t.Callable[[t.Optional[str]],
+                          t.Union[Bookmarks, t.Awaitable[Bookmarks]]]
```

Comment:
since you're in the async bookmark manager, shouldn't it always return `t.Awaitable[Bookmarks]`? why the union here?

Reply:
You'd think so :D I thought I'd give users the option to pass me either an async supplier or a sync one, should they not need any IO. See

https://github.com/neo4j/neo4j-python-driver/pull/771/files#diff-8c202fdf265c9518cf5a49a411975ed9a0942679565aed8651ad9f64261c32e3R157-R158

you can also configure the async driver with an `AsyncBMM` or a `BMM`. I also went down this route with the custom resolver function: it *can* be async (when using the async driver), but doesn't have to.

Reply:
I see, you want to be friends with users, don't you? ;D